src/compiler/factory.ts

```diff
@@ -155,141 +129,166 @@ namespace ts {
         return name;
     }
 
-    export function createLoopVariable(location?: TextRange): Identifier {
-        const name = <Identifier>createNode(SyntaxKind.Identifier, location);
-        name.text = "";
-        name.originalKeywordKind = SyntaxKind.Unknown;
+    /** Create a unique temporary variable for use in a loop. */
+    export function createLoopVariable(): Identifier {
+        const name = createIdentifier("");
         name.autoGenerateKind = GeneratedIdentifierKind.Loop;
         name.autoGenerateId = nextAutoGenerateId;
         nextAutoGenerateId++;
         return name;
     }
 
-    export function createUniqueName(text: string, location?: TextRange): Identifier {
-        const name = <Identifier>createNode(SyntaxKind.Identifier, location);
+    /** Create a unique name based on the supplied text. */
+    export function createUniqueName(text: string): Identifier {
+        const name = createIdentifier("");

```

Comment:
We don't want to call `escapeIdentifier` on `text`, which is the default behavior of `createIdentifier`.